modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Monad.kt

```diff
@@ -50,9 +50,12 @@ interface Monad<F> : Selective<F> {
   fun <A> Kind<F, Kind<F, A>>.flatten(): Kind<F, A> =
     flatMap(::identity)
 
-  fun <A, B> Kind<F, A>.followedBy(fb: Kind<F, B>): Kind<F, B> =
+  override fun <A, B> Kind<F, A>.followedBy(fb: Kind<F, B>): Kind<F, B> =
     flatMap { fb }
 
+  override fun <A, B> Kind<F, A>.apIgnore(fb: Kind<F, B>): Kind<F, A> =
+    fb.flatMap { this }
+
```

Comment:
This is no good! This reverses the order! I think this is also something we already have: `flatTap` does the same right?

Reply:
Not exactly because `flatTap` has a `f: (A) -> Kind<F, B>`. In this case we ignore the value of the monad completely. I made this implementation looking at the overriden version of `ap` for Monad (Line 48)

Reply:
Thats true the signature is different. :+1:  

But still this override is no good as it changes the behaviour:
`IO { println(1) }.apIgnore(IO { println(2) })` will print 1 and then 2 with the apply instance, but 2 and then 1 with the monad instance. 

Just implement it as `flatTap { fb }` which should work.

Reply:
I guess we should somehow try to enforce the monad-applicative consistency law by adding a test to the MonadLaws that takes both a Monad and an explicit Applicative instance and checks if they do the same thing. Will open a ticket for that, as I find this very important!

Reply:
https://github.com/arrow-kt/arrow/issues/1847 Added this as a ticket.